[WIP] Add service form

kane-c opened the pull request:
Content and styles will need revision. Thoughts on this approach? It's not ideal but the form was built with jQuery and Bootstrap in mind.

kane-c committed:
Allow webpack dev server URL to be set in environment

kane-c committed:
Add initial 'add service' form

Content and styles will need revision

DanielHeath commented:
I thought we were linking to a totally other app to do this (in django or something)

DanielHeath commented on src/pages/AddServicePage.js:
```diff
@@ -0,0 +1,101 @@
+ *
+ * @returns {Promise<Object>} a promise for the request.
+ */
+function _request(obj: XhrOptions) {
```
Probably better to use the existing function rather than copying it from iss.js

DanielHeath commented on src/pages/AddServicePage.js:
```diff
@@ -0,0 +1,101 @@
+    }
+
+    async componentDidMount(): void {
+        let response = await _request({
```
Oh, I see - we're ajax-ing the form in.

DanielHeath replied:
`https://iss-staging.infoxchangeapps.net.au/add-service-form?form=ask-izzy` doesn't do anything yet, right?

DanielHeath replied:
Wouldn't an iframe be much easier?

kane-c replied:
Perhaps. That was one of the options we considered, however the problem was that it won't inherit the styles of the page - e.g. custom fonts, etc.

I'll investigate an iframe version a bit more.

DanielHeath replied:
I _really_ don't want to be styling the form part-in-izzy and part-in-iss.

kane-c commented:
Updated. Now uses an iframe instead of hacky Ajax magic.

The only problem now is that the form just uses default Bootstrap styles.

I've added the ability to pass in a href to your own style sheet (`css` parameter on `add-service-form`, but a new stylesheet is required for that.

DanielHeath commented:
++
